### ISSUE TITLE
Improve quadratic_expansion

### ISSUE THREAD

**Issue by schillic** [JuliaReach/IntervalMatrices.jl#10]
Here are some optimizations for the function `quadratic_expansion` (below):
* Swap iteration over `i`/`j` (iterate over columns in Julia).
* Compute `t^2/2` only once.
* Compute `S` first (avoids double access to each cell).
* Inline `k ≠ i` conditions (have two loops from 1 to `i-1` and from `i+1` to `n`) (`if`s in loops are performance killers); the version for `k ≠ i && k ≠ j` needs three loops.
  I think there was some neat Julia trick to make this elegant, but I forgot; the body can be written as a macro to avoid code duplication.
* Generalize to numeric type of `A` (no performance improvement, but nice to have).

https://github.com/JuliaReach/IntervalMatrices.jl/blob/a0247824cc9a907259a0a00d6c55275fc3bc5a74/src/exponential.jl#L17-L56
